bin/liquidator/src/rebalance.rs

```diff
@@ -423,6 +409,301 @@ impl Rebalancer {
         Ok(())
     }
 
+    async fn dust_threshold_for_limit_order(&self, token: &TokenContext) -> anyhow::Result<u64> {
+        let (_, market) = self
+            .mango_client
+            .context
+            .serum3_markets
+            .iter()
+            .find(|(_, context)| {
+                context.base_token_index == token.token_index
+                    && context.quote_token_index == QUOTE_TOKEN_INDEX
+            })
+            .ok_or(anyhow::format_err!(
+                "could not find market for token {}",
+                token.name
+            ))?;
+
+        Ok(market.coin_lot_size - 1)
+    }
+
+    async fn unwind_using_limit_orders(
+        &self,
+        account: &Box<MangoAccountValue>,
+        token: &TokenContext,
+        token_price: I80F48,
+        dust_threshold: FixedI128<U48>,
+        native_amount: I80F48,
+    ) -> anyhow::Result<()> {
+        if native_amount >= 0 && native_amount < dust_threshold {
+            return Ok(());
+        }
+
+        let (market_index, market) = self
+            .mango_client
+            .context
+            .serum3_markets
+            .iter()
+            .find(|(_, context)| {
+                context.base_token_index == token.token_index
+                    && context.quote_token_index == QUOTE_TOKEN_INDEX
+            })
+            .ok_or(anyhow::format_err!(
+                "could not find market for token {}",
+                token.name
+            ))?;
+
+        let side = if native_amount < 0 {
+            Serum3Side::Bid
+        } else {
+            Serum3Side::Ask
+        };
+
+        let distance_from_oracle_price_bp =
+            I80F48::from_num(self.config.limit_order_distance_from_oracle_price_bps)
+                * match side {
+                    Serum3Side::Bid => 1,
+                    Serum3Side::Ask => -1,
+                };
+        let price_adjustment_factor =
+            (I80F48::from_num(10_000) + distance_from_oracle_price_bp) / I80F48::from_num(10_000);
```

Comment:
personally I prefer to store these things as a fraction in f64, instead of bps, e.g. `1.0 + relative_distance_from_oracle_price` - but this is ok too